src/simulations/modeling/cobra_helpers.py

```diff
@@ -159,3 +163,41 @@ def _query_item(item, query_id, query_namespace):
                 if query_id.lower() == annotation.lower():
                     return True
     return False
+
+
+def parse_bigg_compartment(metabolite_id, model):
+    """
+    Parse the compartment ID of the given metabolite BiGG identifier.
+
+    Parameters
+    ----------
+    metabolite_id: str
+        A metabolite identifier, with the bigg convention of appending an underscore
+        followed by the compartment id. For example: "h2o_c"
+    model: cobra.Model
+        The model the metabolite belongs to. Used to confirm that the parsed compartment
+        id exists in the model.
+
+    Returns
+    -------
+    tuple(str, str)
+        First string: The metabolite identifier without its compartment postfix.
+        Second string: The compartment identifier.
+
+    Raises
+    ------
+    ValueError
+        If the given identifier does not match the expected BiGG format, i.e., does not
+        contain an underscore.
+    CompartmentNotFound
+        If the compartment identifier parsed out of the metabolite does not exist in the
+        model.
+    """
+    if "_" not in metabolite_id:
```

Comment:
Instead of the checking for the presence of an underscore, you can simply attempt the split and modify the exception:

```py
try:
    metabolite_id, compartment_id = metabolite_id.rsplit("_", 1)
except ValueError as error:
    raise ValueError(f"The identifier {metabolite_id} is not valid BiGG format.") from error
```

Reply:
Also, I'd change the message to `"The identifier {metabolite_id} has no valid BiGG compartment suffix."`.